core/templates/dev/head/pages/signup/Signup.js

```diff
@@ -143,11 +143,28 @@ oppia.controller('Signup', [
             'Invalid value for email preferences: ' + canReceiveEmailUpdates);
         }
       }
-
       siteAnalyticsService.registerNewSignupEvent();
 
       $scope.submissionInProcess = true;
       $http.post(_SIGNUP_DATA_URL, requestParams).then(function() {
+        var returnUrl = window.decodeURIComponent(
+          urlService.getUrlParams().return_url);
+
+
+        // If the user has the return url as the creator dashboard, it is
```

Comment:
Why do you not do this in the backend POST request, to avoid a second RPC and a potential race condition?